xfel/merging/application/statistics/deltaccint.py

```diff
@@ -0,0 +1,190 @@
+from __future__ import absolute_import, division, print_function
+from six.moves import range
+from dxtbx import flumpy
+from dials.array_family import flex
+import numpy as np
+from scitbx.math import five_number_summary
+from xfel.merging.application.reflection_table_utils import reflection_table_utils
+from xfel.merging.application.worker import worker
+
+
+class deltaccint(worker):
+  '''Calculates ΔCC½ using the σ-τ method from Assmann 2016'''
+
+  def __repr__(self):
+    return 'ΔCC½ statistics (σ-τ method)'
+
+  def run(self, experiments, reflections):
+    self.logger.log_step_time("STATISTICS_DELTA_CCINT")
+
+    comm = self.mpi_helper.comm
+    MPI = self.mpi_helper.MPI
+
+    assert experiments == None, "Must be run after the group worker"
+    # need at least 3 reflections to keep multiplicty > 2 when removing an image
+    filtered = flex.reflection_table()
+    min_mult = max(3, self.params.merging.minimum_multiplicity)
+    for refls in reflection_table_utils.get_next_hkl_reflection_table(reflections=reflections):
+      if len(set(refls['id'])) >= min_mult:
+        filtered.extend(refls)
+
+    # Get list of all experiment identifiers
+    expt_ids = comm.gather(filtered.experiment_identifiers().values(), 0)
+    if self.mpi_helper.rank == 0:
+      all_expt_ids = flex.std_string()
+      for expt_ids_ in expt_ids:
+        all_expt_ids.extend(expt_ids_)
+      expt_ids = list(set(all_expt_ids))
+    else:
+      expt_ids = None
+    all_expt_ids = comm.bcast(expt_ids, 0)
+    all_expts_map = dict(zip(all_expt_ids, range(len(all_expt_ids))))
```

Comment:
This is mostly a matter of style, but I like to write these reverse-maps as dict comprehension: `all_expts_map = {v: k for k, v in enumerate(all_expt_ids)}`.